modules/age.nix

```diff
@@ -201,6 +212,39 @@ in
         "agenixChownKeys"
       ];
     };
+
+    systemd.services = lib.mkMerge
+      (lib.mapAttrsToList
+        (name: {action, services, file, path, mode, owner, group, ...}:
+          let
+            fileHash = builtins.hashFile "sha256" file;
+            restartTriggers = [ fileHash path mode owner group ];
+          in
+            lib.mkMerge [
+              (lib.genAttrs services (_: { inherit restartTriggers; }))
+              (lib.mkIf (action != "") {
+                "agenix-${name}-action" = {
+                  inherit restartTriggers;
+
+                  # We execute the action on reload so that it doesn't happen at
+                  # startup. The only disadvantage is that it won't trigger the
+                  # first time the service is created.
+                  reload = action;
+                  reloadIfChanged = true;
```

Comment:
I think it makes more sense to also trigger at startup. This service is then essentially a secret post-processing-after-loading script.

Reply:
I'm not sure what you mean by "post-processing-after-loading script". This script doesn't do any post-processing, it just restarts the service. If I understand correctly, having it trigger at startup would mean that the dependent service (e.g. wireguard) would start up twice every time you booted, without any good reason.